Charts

chakti opened the pull request:
![screencapture-localhost-4200-charts-1510425368354](https://user-images.githubusercontent.com/11758660/32692395-01c44f92-c720-11e7-9248-792276e5fc44.png)

chakti committed:
Update dependencies

chakti committed:
Add simple routing

chakti committed:
Reorganize charts to separate module

chakti committed:
disable router tracing

chakti committed:
Add line chart component

chakti committed:
Add bar chart component

chakti committed:
Add missing styles for bar chart

chakti committed:
Add pie chart component

chakti committed:
Add simple navigation

chakti committed:
Add bar chart service

chakti committed:
Add line chart service

chakti committed:
Add pie chart service

Drapegnik commented on src/app/app-routing.module.ts:
```diff
@@ -0,0 +1,20 @@
+import { NgModule }              from '@angular/core';
+import { RouterModule, Routes }  from '@angular/router';
+ 
+import { AppComponent } from './app.component';
```
how we can use absolute imports?, like `app/app.component`

Drapegnik commented on src/app/charts/bar-chart/bar-chart.component.ts:
```diff
@@ -0,0 +1,50 @@
+  selector: 'bar-chart',
+  templateUrl: './bar-chart.component.html',
+  styleUrls: [
+    '../../../../node_modules/nvd3/build/nv.d3.css',
```
looks wrong :worried:

Drapegnik replied:
i think this file should be included once in `.angular-cli.json`, in `apps.styles` config

Drapegnik commented on src/app/charts/charts-routing.module.ts:
```diff
@@ -0,0 +1,18 @@
+import { NgModule }              from '@angular/core';
```
a lot of redundant spaces

Drapegnik commented on src/app/charts/line-chart/line-chart.component.ts:
```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
```
is this code  generated? mb delete all tests and other redundant code to make project more clean and readable?

Drapegnik reviewed (commented):
charts components have a lot of common code, i think it'll make sense combine them

chakti committed:
Remove all spec

chakti committed:
Remove all unused code

chakti committed:
Combine charts components

chakti committed:
Remove extra spaces

Drapegnik reviewed (approved):
👍

Commit:
Add a few improvements

- Remove global d3 imports
- Move all assets to assets/ folder
- Add @types/d3 and move all typings to devDependencies

jakwuh commented:
Запушил то, что заметил.
Мне не хватило настроек tslint чтобы IDEA автоформатировала (e.g. charts.service.ts)

jakwuh commented:
e2e тесты теперь падают

Commit:
Update e2e test

Commit:
Use Headless browsers

jakwuh commented:
Для тестов теперь headless браузеры, @Drapegnik можно подключить travis ci в два клика

jakwuh commented:
Еще вопрос по этому блоку кода:

```js
return Observable.of(chartsMock.pieChartData);
```

Сейчас это просто class property, то есть Observable здесь бесполезен. Я правильно понимаю, что это сделано для того, чтобы в будущем сделать `pieChartData` Observable и все заработает as-is?

chakti commented:
@jakwuh Да, потом эти данные диагармм будут браться с сервера.
